### DISCUSSION TITLE
Cached Local Library Listing

### DISCUSSION THREAD

**Discussion by mathieugoodman** [vicwomg/pikaraoke#205] [Ideas]
Hello,

I have a large local library of MP3+G files and it takes a few minutes for PiKaraoke to read it when loading. Since my library rarely changes, would it be possible to either preload the list as a file or have it cached somehow in a way that it doesn't need to rescan the entire directory every time I start up PiKaraoke?

Thanks!

**Comment by RecursiveFunctions** [comment]
Maybe one way to go about this is like so:
```
ls > ~/pikaraoke-songs/cache
```

I'd have to take a closer look under the hood to understand how pikaraoke is getting all the songs

**Reply by LouOtway** [reply]
I'm in a similar position, have about 90k tracks locally, they will likely never be updated and so it's a shame to build that list time and again on startup.

The code to build the local song list is here:
https://github.com/vicwomg/pikaraoke/blob/0741c2c99ed7053e11da2767488bb88b66faebe9/karaoke.py#L306

To cache this between sessions you'd need to make some kind of list of tracks, maybe as a simple flat file or as a JSON object and load it. 

Perhaps a new command line argumnent could be added to point at an existing tracklist and, if it exists, to just use it?

**Comment by safepay** [comment]
Resolved with #792 once merged.

**Comment by safepay** [comment]
Resolved via #792
